ownCloud/Client/Tasks/InstantMediaUploadTaskExtension.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  InstantMediaUploadTaskExtension.swift
+//  ownCloud
+//
+//  Created by Michael Neuwert on 24.07.2019.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+/*
+* Copyright (C) 2018, ownCloud GmbH.
+*
+* This code is covered by the GNU Public License Version 3.
+*
+* For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+* You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+*
+*/
+
+import Foundation
+import ownCloudSDK
+import Photos
+
+class InstantMediaUploadTaskExtension : ScheduledTaskAction, OCCoreDelegate {
+
+	enum MediaType {
+		case images, videos, imagesAndVideos
+	}
+
+	override class var identifier : OCExtensionIdentifier? { return OCExtensionIdentifier("com.owncloud.action.instant_media_upload") }
+	override class var locations : [OCExtensionLocationIdentifier]? { return [.appDidComeToForeground] }
+	override class var features : [String : Any]? { return [ FeatureKeys.photoLibraryChanged : true, FeatureKeys.runOnWifi : true] }
+
+	override func run(background:Bool) {
+		guard let userDefaults = OCAppIdentity.shared.userDefaults else { return }
+
+		guard userDefaults.instantUploadPhotos == true || userDefaults.instantUploadVideos == true else { return }
+
+		guard let bookmarkUUID = userDefaults.instantUploadBookmarkUUID else { return }
+
+		guard let path = userDefaults.instantUploadPath else { return }
+
+		if let bookmark = OCBookmarkManager.shared.bookmark(for: bookmarkUUID) {
+
+			OCCoreManager.shared.requestCore(for: bookmark, setup:nil, completionHandler: { (core, _) in
+				if let core = core {
+					core.delegate = self
+
+					let pathQuery = OCQuery(forPath: path)
+					pathQuery.changesAvailableNotificationHandler = { query in
+						if query.state == .idle {
+							core.stop(query)
```

Comment:
Starting a query will trigger a full `PROPFIND` for the directory, which could cause quite some traffic if there are already many files in that directory.

You can avoid that overhead by using `-[OCCore trackItemAtPath:(OCPath)path trackingHandler:(void(^)(NSError *error, OCItem *item, BOOL isInitial))trackingHandler]`  instead. If there already is an `OCItem` for that path, it'll return it from cache immediately, without making a potentially [CPU/memory/time]-expensive `PROPFIND`.